pkg/clicmd/root.go

```diff
@@ -122,17 +131,24 @@ func NewRoot(appFs afero.Fs, wd string, args []string) (*cobra.Command, error) {
 	var a app.App
 	var err error
 
-	cmdName, err := parseCommand(args)
+	cmdName, hasHelpFlag, err := parseCommand(args)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(args) > 0 && cmdName != "init" {
+	cmds := []string{"init", "version", "help"}
+	switch {
+	// Commands that do not require a ksonnet application
+	case strings.InSlice(cmdName, cmds) || hasHelpFlag:
```

Comment:
`case` supports a list of cases which are effectively ORed, so another way to say the same thing might be:

```
case strings.InSlice(cmdName, cmds), hasHelpFlag:
```

Either way is fine.
See: https://golang.org/doc/effective_go.html?#switch